README.md
# Digital Paper Edit - React Components Catalogue

[The Storybook](https://bbc.github.io/digital-paper-edit-storybook/?path=/story/breadcrumb--projects)

---> _Work in progress_ <--

**! What is published in the Storybook may be different to what is being actively used in Digital-Paper-Edit-Client, as the components are currently being migrated. !**

This repository uses [storybook](https://storybook.js.org) to gather the components of the React front end. You can read about the "why" in [this ADR](https://github.com/bbc/digital-paper-edit-storybook/blob/9755e2c62343decf05ec87b9c7d85678d58e2186/docs/ADR/reason-09-11.md).

See [here](https://github.com/bbc/digital-paper-edit-client#project-architecture) for the overall project architecture.

## Setup

<!-- _stack - optional_
_How to build and run the code/app_ -->

```sh
git clone git@github.com:bbc/digital-paper-edit-storybook.git
```

```sh
cd digital-paper-edit-storybook
```

Optional step to setup [nvm](https://github.com/nvm-sh/nvm) to use node version 10, otherwise just use node version 10

```sh
nvm use || nvm install
```

in root of project

```sh
npm install
```

## Usage - development

Start the storybook

```sh
npm start
```

[Storybook](https://storybook.js.org/) is at [`http://localhost:6006/`](http://localhost:6006)

### Linking storybook to the Firebase app for development

In this repo, run:
```sh
npm run link:setup
```

In another terminal window:
```sh
npm run link
```

When finished, run:
```sh
npm run unlink
```

## Usage - production

```sh
npm install @bbc/digital-paper-edit-storybook
```

<!-- Example of import

```js
import 'Breadcrumb' from '@bbc/digital-paper-edit-storybook/Breadcrumb'

<Breadcrumb />
```

 -->

## System Architecture

<!-- _High level overview of system architecture_ -->

- uses [`storybook`](https://storybook.js.org) with the setup as [explained in their docs](https://storybook.js.org/docs/guides/guide-react/) to develop this React.
  <!-- - This uses [CSS Modules](https://github.com/css-modules/css-modules) to contain the scope of the css for this component. -->
- [.storybook/webpack.config.js](./.storybook/webpack.config.js) changes the storybook webpack config to add support for css modules.
  <!-- - The parts of the component are inside [`./packages`](./packages) -->
  <!-- - [babel.config.js](./babel.config.js) provides root level system config for [babel 7](https://babeljs.io/docs/en/next/config-files#project-wide-configuration). -->

## Development env

 <!-- _How to run the development environment_
_Coding style convention ref optional, eg which linter to use_
_Linting, github pre-push hook - optional_ -->

- npm > `6.1.0`
- [Node 10 - dubnium](https://scotch.io/tutorials/whats-new-in-node-10-dubnium)

Node version is set in node version manager [`.nvmrc`](https://github.com/creationix/nvm#nvmrc)

## Build

<!-- _How to run build_ -->

To transpile `./src` and create a build in the `./dist` folder, run:

```sh
npm run build
```

## Demo & storybook

- **Storybook** can bew viewed at [https://bbc.github.io/digital-paper-edit-storybook/](https://bbc.github.io/digital-paper-edit-storybook/)
- **Demo** can be viewed at [https://bbc.github.io/digital-paper-edit-client](https://bbc.github.io/digital-paper-edit-client/)

## Build - storybook

To build the storybook as a static site

```sh
npm run build-storybook
```

## Publish storybook & demo to github pages

This github repository uses [github pages](https://pages.github.com/) to host the storybook and the demo of the component. **Make sure to commit and push any changes to MASTER before deploying to github pages.**

```sh
npm run publish-ghpages
```

Alternatively, if you simply want to build the demo locally in the `build` folder then just

```sh
npm run build-storybook
```

## Tests

<!-- _How to carry out tests_ -->

Test coverage using [`jest`](https://jestjs.io/), to run tests

```sh
npm run test
```

## Travis CI

On commit, - this repo uses the [.travis.yml](./.travis.yml) config to run the automated test on [travis CI](https://travis-ci.org/bbc/digital-paper-edit-storybook).

## Deployment

### NPM

> Note that only `README.md` and the `dist` folders are published to NPM.

#### New releases

1. Go into the `release` branch and pull the new changes from `master`.
2. Commit or push changes to the remote branch of `release`
3. Find corresponding changes in [Travis](https://travis-ci.org/github/bbc/digital-paper-edit-storybook)

If you had to make changes in `release`, make sure that the `release` is equivalent to `master`!

#### Locally, using the NPM cli

To push to [npm - `@bbc/digital-paper-edit-storybook`](https://www.npmjs.com/package/@bbc/digital-paper-edit-storybook)

login into npm

```bash
npm run publish-public
```

This runs `npm run build` and `npm publish --access public` under the hood

## Contributing

See [CONTRIBUTING.md](./CONTRIBUTING.md) guidelines and [CODE_OF_CONDUCT.md](./CODE_OF_CONDUCT.md) guidelines.

## Licence

<!-- mention MIT Licence -->

See [LICENCE](./LICENCE.md)

## Legal Disclaimer

_Despite using React, the BBC is not promoting any Facebook products or other commercial interest._


CHANGELOG.md
# Changelog

The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).

## [Released]

## [ 1.29.0 ] - 2021-07-08
Migrate publishing to npm from Travis -> GitHub Actions

## [ 1.28.1 ] – 2021-07-06
- I219 added analytics

## [ 1.27.0 ] - 2021-06-19
- I105 Updates TranscriptRow components

## 1.26.0
- I92 Updates placeholder text and titles of project, transcript and programme script modals

## [ 1.25.0 ]
- I230 Updated design of the workspace: TranscriptRow component

## [ 1.24.0 ]
- I222 Updated design of the workspace: List, ProjectRow and TranscriptRow components

## [ 1.23.2 ] - 2021-05-12
- I91 Removed postinstall script in the dist folder

## [ 1.23.1 ] - 2021-05-11
- I91 Replaced postinstall script in the dist folder

## [ 1.23.0 ] - 2021-05-10
- I72 Added npm link scripts, babel plugin and react-dom dependency

## [ 1.22.2 ] - 2021-04-20
- I77 Updated travis.yml to replace broken Jenkins npm credentials with team member credentials

## [ 1.22.1 ] - 2021-04-20
- I77 Updated travis.yml to replace Eimi's npm credentials with our Jenkins credentials







src/dummy.js
const items = [
  {
    id: '1234',
    projectId: '5678',
    key: 'abc123',
    title: 'Sample Simple Card Title One',
    description: 'This is a sample card description. This is fun!',
    display: true,
    created: { seconds: 1576057570, nanoseconds: 390000000 },
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    url: '/projects/1/transcripts/1234'
  },
  {
    id: '5678',
    projectId: '5678',
    key: 'def456',
    title: 'Sample Simple Card Title Two',
    description: 'This is a sample card description. This is fun!',
    display: true,
    created: { seconds: 1576057570, nanoseconds: 390000000 },
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    url: '/projects/1/transcripts/5678',
    mediaType: 'audio'
  }
];

const transcriptItems = [
  {
    id: '1',
    key: 'transcript_key_1',
    title: 'Title - Done Transcript',
    description: 'This is a sample card description. This is fun!',
    url: '/projects/1/transcripts/',
    projectId: '1',
    created: '2021-05-19, 10:53',
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    status: 'done',
    display: true,
    mediaType: 'audio'
  },
  {
    id: '3',
    key: 'transcript_key_2',
    title: 'Title - In Progress Transcript',
    description: 'This is a sample card description. This is fun!',
    url: '/projects/1/transcripts/',
    projectId: '1',
    created: '2021-03-19, 08:03',
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    status: 'in-progress',
    display: true,
    mediaType: 'video'
  },
  {
    id: '4',
    key: 'transcript_key_3',
    title: 'Title - Error Transcript',
    description: 'This is a sample card description. This is fun!',
    url: '/projects/1/transcripts/',
    status: 'error',
    message: 'Something has gone wrong with your transcription',
    projectId: '1',
    created: '2021-05-02, 14:53',
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    display: true,
    mediaType: 'audio'
  },
  {
    id: '5',
    key: 'transcript_key_4',
    title: 'Title - Uploading Transcript',
    description: 'This is a sample card description. This is fun!',
    url: '/projects/1/transcripts/',
    message: 'warning message',
    display: true,
    progress: 100,
    projectId: '1',
    created: '2021-05-19, 19:03',
    updated: { seconds: 1576057575, nanoseconds: 390000000 },
    status: 'uploading',
    mediaType: 'video'
  }
];

const modalItems = [
  {
    id: '1',
    type: 'project',
    showModal: true,
    title: 'Example Project Title',
    description: 'This is a sample card description. This is fun!',
    url: '/projects/1/transcripts/1234',
    modalTitle: 'Edit Project'
  },
  {
    showModal: true,
    modalTitle: 'New programme script',
    id: '2',
    type: 'programme-script'
  },
  {
    projectId: '123',
    title: '',
    description: '',
    uploadCompleted: true,
    showModal: true,
    modalTitle: 'New Transcript',
    id: '3',
    type: 'transcript'
  },
  {
    projectId: '1234',
    title: '',
    description: '',
    uploadCompleted: true,
    showModal: false,
    modalTitle: 'New Transcript',
    id: '4',
    type: 'transcript'
  }
];

const programmeScript = [
  {
    type: 'title',
    text: 'An immense Achievement'
  },
  {
    type: 'paper-cut',
    id: 1,
    speaker: 'Mr Loud',
    words: [
      {
        text: 'Greatest day of my life was when I wrote this text.',
        start: 0,
        end: 1
      }
    ]
  },
  {
    type: 'note',
    text: 'Maybe a little bit obnoxious'
  },
  {
    type: 'insert',
    text: 'Insert New Selection here'
  },
  {
    type: 'paper-cut',
    id: 2,
    speaker: 'Mrs Loud',
    words: [
      {
        text: 'Greatest day of my life was when I spoke this text.',
        start: 0,
        end: 1
      }
    ]
  },
  {
    type: 'voice-over',
    text: 'link: wonderful times of the Loud family'
  }
];

export { items, transcriptItems, modalItems, programmeScript };
